modules/graphics/labwc.nix

```diff
@@ -0,0 +1,142 @@
+# Copyright 2022-2023 TII (SSRC) and the Ghaf contributors
+# SPDX-License-Identifier: Apache-2.0
+{
+  lib,
+  pkgs,
+  config,
+  ...
+}: let
+  cfg = config.ghaf.graphics.labwc;
+in {
+  options.ghaf.graphics.labwc = {
+    enable = lib.mkEnableOption "labwc";
+  };
+
+  config = lib.mkIf cfg.enable {
+    environment.systemPackages = with pkgs; [
+      # Seatd is needed to manage log-in process for wayland sessions
+      seatd
+      labwc
+      foot
+      swaybg
+      kanshi
+      waybar
+      mako
+      swayidle
+    ];
+
+    # Next 4 services/targets are taken from official labwc documentation:
+    # https://wayland.pages.freedesktop.org/labwc/toc/running-labwc.html
+
+    # labwc socket
+    systemd.user.sockets."labwc" = {
+      unitConfig = {
+        Description = "labwc, a Wayland compositor";
+        Documentation = "man:labwc(1)";
+      };
+      socketConfig = {
+        ListenStream = "%t/wayland-0";
+      };
+      wantedBy = ["labwc.service"];
+    };
+
+    # labwc service
+    systemd.user.services."labwc" = {
+      enable = true;
+      description = "labwc, a Wayland compositor, as a user service TEST";
+      documentation = ["man:labwc(1)"];
+      requires = ["labwc.socket"];
+      after = ["labwc.socket" "ghaf-session.service"];
+      serviceConfig = {
+        # Previously there was "notify" type, but for some reason
+        # systemd kills labwc.service because of timeout (even if it is disabled).
+        # "simple" works pretty well, so let's leave it.
+        Type = "simple";
+        #TimeoutStartSec = "60";
+        #WatchdogSec = "20";
+        # Defaults to journal
+        StandardOutput = "journal";
+        StandardError = "journal";
+        ExecStart = "${pkgs.labwc}/bin/labwc -C ${pkgs.labwc}/share/doc/labwc -s ${pkgs.labwc}/share/doc/labwc/autostart";
+        #GPU pt needs some time to start - labwc fails to restart 3 times in avg.
```

Comment:
I think this is ok for the first iteration to enable labwc but I'd like to ask you to elaborate "GPU pt needs some time to start" for further development.
 
What I'm after is that if "GPU pt" refers to GPU driver loading or some GPU related systemd service unit defined elsewhere, we could maybe wait for that service unit and add it to `after  = [ "gpu-pt-or-something.service"]` instead of restart-sleep-until-succeed? As said, that can be further development but please refine the comment of "GPU pt needs ..." for future readers.

Reply:
I'm actually also curious what causes this, earlier I added i915 to the initrd of the guivm to ensure it is loaded early during boot

Reply:
The issue SP-2838 is created to track this issue, all the info about that is there.